### ISSUE TITLE
Implement “Connect GA” CTA after Key Metrics widget setup with GA4 disconnected/unavailable

### ISSUE THREAD

**Issue by jimmymadon** [google/site-kit-wp#6265]
## Feature Description

If more than two of the selected metrics within the Key Metrics widget area are from GA4, but GA4 is disconnected, then a new CTA banner should be rendered to encourage the user to reconnect the GA.

Refer to the [relevant section](https://docs.google.com/document/d/1k8NOo2xEHUQAcgYG0TB5sUjGcPiC8SWIalEdd9G8-9g/edit#heading=h.hfwitp2lb9g3) in the design doc for additional context.

![Screenshot 2022-12-14 at 12 20 53](https://user-images.githubusercontent.com/1721319/207593908-a1dbf7f6-b71f-4b5e-a80c-0104cf15c004.png)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* A new Connect GA CTA banner should be implemented as per the [Figma mock](https://www.figma.com/file/CdtrHB0HAjfGRWTnTMMQI0/Key-Metrics?node-id=1766%3A47746&t=bwOtTQ7S3Odzwt7u-0). ~Note that the width of the tile should be the same as a widget's standard "quarter" width and not 1/3rd as in the Figma design.~
* The banner should be rendered if more than 2 of the selected metrics are dependant on GA4 and GA4 was disconnected after the widget's setup.
* Clicking on the button in the CTA should take the user to the "Connect service - Analytics" page to setup GA (Similar to the "Setup Google Analytics" CTA within the Search Traffic widget.

#### Create the components

* Wait for #6209, #6257, and #6313 are merged.
* Create a new `ConnectGA4CTA` component in the same directory #6209 is implemented.
* The component should render the following:
  * Start with a `section` tag. Followed by `Grid->Row->Cell` components.
  * Render the red heading text within an `h3` tag per the Figma design. The text should be translated.
  * Render the description text within a `p` tag per the Figma design. The text should be translated.
  * Render the `Connect Google Analytics` button within the `Button` component per the Figma design. The text should be translated. The `onClick` callback should follow a similar implementation as the `SetupModule` component:
  https://github.com/google/site-kit-wp/blob/43c436489a48caeead4c33809187b6823fb0aaaf/assets/js/components/settings/SetupModule.js#L57-L82
  * Extract the SVGs from the Figma design and render them. It should match Figma design.
  ![Screenshot 2023-01-09 at 1 30 15 PM](https://user-images.githubusercontent.com/24408261/211267539-bf9a127e-2921-430d-a7e9-cb149354d93e.png)
* Create a footer element or a reusable child component with the following:
  * Start with a `footer` tag. Followed by `Grid->Row->Cell` components.
  * Render the `Maybe later` link within the `Link` component per the Figma design. The text should be translated.
  * The `onClick` callback should dismiss the CTA by dispatching the `dismissItem` action. An appropriate constant key should be created to pass it to the action.
* The component should render `null` if the GA4 module is connected using the `isModuleConnected` selector.
* Using the `getKeyMetrics` selector, the component should check if any of the selected metrics are from GA4. If so, it should render the `CTA` component. Otherwise, it should render `null`.
* Create a new `ConnectGA4CTAWidget` component in the same directory #6209 is implemented.
* It takes the `Widget` and `WidgetNull` components as props.
* Render the `ConnectGA4CTA` component within the `Widget` component if the following conditions are met:
  * If the GA4 connect CTA is not dismissed. Check using the `isItemDismissed` selector.
  * If the GA4 module isn't connected. Check using the `isModuleConnected` selector.
  * If the selected metrics are dependent on GA4. Check using the `getKeyMetrics` selector.
* Otherwise, render the `WidgetNull` component.
* Create stories for the `ConnectGA4CTAWidget` component.

### Register the widget

* Register the `ConnectGA4CTAWidget` component in the `KeyMetrics` widget area once #6313 is merged.
* The widget should be registered with the following settings:
  * `width` should be `FULL`.
  * `wrapWidget` should be `false`.
  * `priority` should be `1`.
  * `modules` should be `['analytics']`.

### Test Coverage

* Add tests for the `ConnectGA4CTA` component.

## QA Brief

- Set up Site Kit with the Analytics module.
- Enable the `userInput` feature flag.
- Complete the user input questionnaire and answer `Publish a blog` for the first (purpose) question. This should configure key metrics to display widgets all dependant on Analytics.
- Go to the Site Kit dashboard and verify that you are able to see key metric widgets.
- Now, from Site Kit settings, disconnect the Analytics module.
- Go back to the Site Kit dashboard and verify that you can see the `Google Analytics is disconnected` CTA widget, according to the ACs and Figma designs.
- Clicking on the CTA should navigate you to Analytics module setup.
- Clicking on `Maybe later` should dismiss the widget.
- Verify the `Key Metrics -> ConnectGA4CTAWidget` Storybook story.

## Changelog entry

* Add CTA to connect Analytics if disconnected after setting up Key Metrics.

**Comment by tofumatt** [comment]
From the Figma mocks, it looks like the CTA widget could be rendered as a half-width widget if it makes sense (eg. with multiple widgets specified, like https://github.com/google/site-kit-wp/blob/c2f085d161824949faa7b0fd0a45e187e1deb50c/assets/js/modules/adsense/index.js#L128).

But that can be addressed in the IB. Anyway, ACs here look good, moving to IB 👍🏻

**Comment by tofumatt** [comment]
All looks good, just one minor naming point; the point of the module is to Connect GA4, so might as well name it `ConnectGA4CTA` instead of `GA4CTAConnect` which just seems a bit less obvious to parse. 😅 

IB ✅

**Comment by nfmohit** [comment]
I'm seeing a discrepancy between the ACs and the IB here regarding the `Connect Google Analytics` CTA action.

The ACs say:

> Clicking on the button in the CTA should take the user to the "Connect services" page to setup GA.

The IB says:

> Render the `Connect Google Analytics` button within the `Button` component per the Figma design. The text should be translated. The `onClick` callback should follow a similar implementation as the `SetupModule` component.

I do not think connecting the module would be as simple as the SetupModule component because there are aspects of selecting an Analytics account, property, web data stream, and error handling.

@jimmymadon Any advice on which approach should be taken here?

CC: @hussain-t @tofumatt

**Comment by nfmohit** [comment]
> @nfmohit I have updated the ACs here. I think we can use the same behaviour as in the "Set up Google Analytics" CTA within the Search Traffic widget. Would this be fine?

That's perfect, thank you @jimmymadon!

**Comment by wpdarren** [comment]
### QA Update: ⚠️ 

@nfmohit this is looking good but I do have a few observations.  

These could be expected at this point in the engineering, but I do not want to assume 😁 

1. I am seeing console errors when I return to the dashboard after disconnecting Analytics. 

`Google Site Kit API Error method:GET datapoint:report type:modules identifier:analytics-4 error:"Module must be active to request data."`

2. On the Figma designs, all of the key metrics tiles disappear, but I still see them but they are set to 0% etc. 

See Screenshot below. 

![image](https://github.com/google/site-kit-wp/assets/73545194/37906c1f-9a3d-46d3-b9f8-ee1f583a4acf)

**Comment by nfmohit** [comment]
Thank you for sharing your observations, @wpdarren!

@jimmymadon Could you possibly help a little to answer Darren's questions here?

> On the Figma designs, all of the key metrics tiles disappear, but I still see them but they are set to 0% etc.

The ACs of this issue don't mention about updating the visibility of the other widgets. Is this something that will be addressed by another issue?

> I am seeing console errors when I return to the dashboard after disconnecting Analytics.

According to my observation, these errors are caused by the other widgets making GA4 report requests. Not rendering them when GA4 is disconnected should solve it.

> If you click on the 'Maybe later' link the CTA does disappear, but when you connect Analytics again, and then disconnect later, the CTA no longer appears. Is this expected? I suspect not.

This is indeed a valid concern. What do you think of setting an expiry to the dismissal so that it expires after a while, say 24 hours? That way, it'll not show up if GA4 is connected again, but will show up again if it is disconnected again.

Thank you!

**Comment by nfmohit** [comment]
> > > If you click on the 'Maybe later' link the CTA does disappear, but when you connect Analytics again, and then disconnect later, the CTA no longer appears. Is this expected? I suspect not.
> 
> > This is indeed a valid concern. What do you think of setting an expiry to the dismissal so that it expires after a while, say 24 hours? That way, it'll not show up if GA4 is connected again, but will show up again if it is disconnected again.
> 
> I feel we should "reset" this persistent dismissal when GA4 is "connected" rather than relying on expiry times. We could perhaps do this in the `on_activation` callback? A merge reviewer can perhaps approve this quickly before we proceed.

That is a good idea. However, if I'm not wrong, I do not think we have in our infrastructure a way to remove a dismissed item. One way could be updating the dismissed item with a very small expiry, e.g. 1 second. Otherwise, we'll have to enhance the `Dismissed_Items` class to allow removing dismissed items.

Additionally, I think it'd be wiser to address this in a new issue as everything else is according to spec.

@jimmymadon What do you think? Thank you!

@eugene-manuilov @aaemnnosttv @tofumatt @techanvil Requesting feedback from one/more of you here too please, thank you!

**Comment by aaemnnosttv** [comment]
> That is a good idea. However, if I'm not wrong, I do not think we have in our infrastructure a way to remove a dismissed item. One way could be updating the dismissed item with a very small expiry, e.g. 1 second. Otherwise, we'll have to enhance the `Dismissed_Items` class to allow removing dismissed items.

It seems reasonable that we should allow for removing a dismissed item. We could technically do it now as you said by re-adding it again with the smallest possible TTL, but that would be unnecessarily indirect to avoid adding a simple method to do the same thing explicitly.

> Additionally, I think it'd be wiser to address this in a new issue as everything else is according to spec.

Agreed, this is outside of the scope and also involves some changes that wouldn't be entirely behind the feature flag so it's best to handle this separately.

**Comment by wpdarren** [comment]
### QA Update: ✅ 

**Verified:**

- Go to the Site Kit dashboard after setting up Site Kit and GA. and are able to see key metric widgets.
- Disconnect the Analytics module. You can see the Google Analytics is disconnected CTA widget.
  - Clicking on the CTA navigates you to Analytics module setup.
  - Clicking on Maybe later dismisses the widget.
- Verified the Key Metrics -> ConnectGA4CTAWidget Storybook story.

Please note that 3 issues were [identified above](https://github.com/google/site-kit-wp/issues/6265#issuecomment-1627860092).  All will be fixed in other tickets. 

<details>
 <summary>Screenshots</summary>

![image](https://github.com/google/site-kit-wp/assets/73545194/e9cdd10a-334c-4637-9759-965e0f5b7015)

![image](https://github.com/google/site-kit-wp/assets/73545194/e413713e-e36c-4e7f-82d0-8d354b1ca483)

</details>
